test/libsolidity/InlineAssembly.cpp

```diff
@@ -228,8 +228,8 @@ BOOST_AUTO_TEST_CASE(switch_statement)
 	BOOST_CHECK(successParse("{ switch 42 default {} }"));
 	BOOST_CHECK(successParse("{ switch 42 case 1 {} default {} }"));
 	BOOST_CHECK(successParse("{ switch 42 case 1 {} case 2 {} default {} }"));
-	BOOST_CHECK(successParse("{ switch mul case 1 {} case 2 {} default {} }"));
-	BOOST_CHECK(successParse("{ switch mul() case 1 {} case 2 {} default {} }"));
+	BOOST_CHECK(successParse("{ 1 2 switch mul case 1 {} case 2 {} default {} }"));
```

Comment:
I think the switch expression should be a functional expression, i.e. cannot take arguments from the stack.